javatests/arcs/android/systemhealth/testapp/TestStorageService.kt

```diff
@@ -0,0 +1,68 @@
+package arcs.android.systemhealth.testapp
+
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import arcs.android.storage.ParcelableStoreOptions
+import arcs.sdk.android.storage.service.StorageService
+import arcs.sdk.android.storage.service.StorageServiceBindingDelegate
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+/**
+ * Arcs system-health-test storage service. Supports crashing itself when needed.
+ */
+class TestStorageService : StorageService() {
+    private val coroutineContext = Dispatchers.Main + CoroutineName("TestStorageService")
+    private val scope = CoroutineScope(coroutineContext)
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        if (intent != null &&
+            intent.hasExtra(EXTRA_CRASH) &&
+            intent.getBooleanExtra (EXTRA_CRASH, false)) {
+            scope.launch {
+                delay(1000)
```

Comment:
Could you make it also random? 
The 1000 ms seems a bit long where all storage service tasks might be already finished prior to the crash.
So we're almost testing its stability in *idle* state.

Reply:
Done.